src/main/java/com/ibm/watson/modelmesh/payload/RemotePayloadProcessor.java

```diff
@@ -58,16 +62,34 @@ private static PayloadContent prepareContentBody(Payload payload) {
         ByteBuf byteBuf = payload.getData();
         String data;
         if (byteBuf != null) {
-            ByteBuf encoded = Base64.encode(byteBuf, byteBuf.readerIndex(), byteBuf.readableBytes(), false);
-            //TODO custom jackson serialization for this field to avoid round-tripping to string
-            data = encoded.toString(StandardCharsets.US_ASCII);
+            data = encodeBinaryToString(byteBuf);
         } else {
             data = "";
         }
+        Metadata metadata = payload.getMetadata();
+        Map<String, String> metadataMap = new HashMap<>();
+        if (metadata != null) {
+            for (String key : metadata.keys()) {
+                if (key.endsWith("-bin")) {
+                    byte[] bytes = metadata.get(Metadata.Key.of(key, Metadata.BINARY_BYTE_MARSHALLER));
+                    if (bytes != null) {
+                        metadataMap.put(key, encodeBinaryToString(Unpooled.wrappedBuffer(bytes)));
```

Comment:
This should be more efficient here, no need to wrap in a ByteBuf since it's already a byte array.
```suggestion
                        metadataMap.put(key, new String(bytes, StandardCharsets.US_ASCII));
```

Reply:
@tteofili oh sorry my bad, you're right, I got mixed up re base64.

But similarly it's probably more efficient here to do (with `java.util.Base64`):
```suggestion
                        metadataMap.put(key, Base64.getEncoder().encodeToString(bytes));
```

Reply:
ok, done, I think we're fine, given it's ok to use both `java.util` and Netty's `Base64` classes here (for handling `byte[]` and `ByteBuf` respectively).